### DISCUSSION TITLE
Make autocoded files more visible

### DISCUSSION THREAD

**Discussion by timcanham** [nasa/fprime#1674] [Call for Comments]
With cmake using build caches and the F Prime cmake system generating the autocoded base classes, a frequent question is: How do I find the generated files?

Possible ideas:

1) Create symlinks in the source directory to the build cache directory.
2) The files aren't really target-specific. Can the generated files go in a different, predictable path in the build cache?

**Comment by LeStarch** [comment]
`fprime-util info` was designed to provide this and other information

```
fprime-util info
[INFO] fprime build information:
    Available directory targets: 'build'

    Available global targets: 'build --all' 'build --all --ut' 'check --all' 'check --leak --all' 'check --coverage --all'
  ----------------------------------------------------------
    Release build cache module directory: /Users/mstarch/code/fprime-infra/fprime-sw/Ref/build-fprime-automatic-native/Ref
    Release build cache: /Users/mstarch/code/fprime-infra/fprime-sw/Ref/build-fprime-automatic-native
```

The first line after `---------...` is the path to the module's cache directory.  Also note, there isn't "target specific" information here.  Is this information insufficient?

**Comment by timcanham** [comment]
Isn't the `build-fprime-automatic-*native*` the target-specific part?

**Comment by timcanham** [comment]
My proposal for 1. was to put a symlink in each directory that has fpp files to the build cache directory with the autocoded files for that fpp file.

**Comment by thomas-bc** [comment]
Yay! Reviving this cause it's important for understanding the framework's inner-working.
